python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+            for sub_schema in src_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_sources (%s)" % name_type_string)
+
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects_to_dia_soruces "
+                "(src_id INTEGER PRIMARY KEY, obj_id INTEGER)")
+
+            self._db_connection.commit()
+
+        return True
+
+    @pipeBase.timeMethod
+    def load(self, ctr_coord, radius):
+        """ Load all DIAObjects and associated DIASources.
+
+        Parameters
+        ----------
+        ctr_coord : lsst.afw.geom.Coord
+            Center position of on on sky circle to load.
+        radius : lsst.afw.geom.Angle
+
+
+        Returns
+        -------
+        lsst.ap.association.DIAObjectCollection
+        """
+        indexer_indices, on_boundry = self.indexer.get_pixel_ids(
+            ctr_coord, radius)
+
+        dia_objects = self.get_dia_objects(indexer_indices)
+
+        dia_collection = DIAObjectCollection(dia_objects)
+
+        return dia_collection
+
+    @pipeBase.timeMethod
+    def store(self, dia_collection, compute_spatial_index=False):
+        """ Store all DIAObjects and DIASources in this dia_collection.
+
+        Parameters
+        ----------
+        dia_collection : lsst.ap.association.DIAObjectCollection
+            Collection DIAObjects to store. Also stores the DIASources
+            associated with these DIAObjects.
+        compute_spatial_index : bool
+            If True, compute the spatial search indices using the
+            indexer specified at class creation.
+        """
+        for dia_object in dia_collection.dia_objects:
+            if compute_spatial_index:
+                dia_object.dia_object_record.set(
+                    'indexer_id', self.indexer.index_points(
+                        [dia_object.ra], [dia_object.dec])[0])
+            self.store_dia_object(dia_object)
+            for dia_source in dia_object.dia_source_catalog:
+                self.store_dia_source(dia_source)
+                self.store_dia_object_source_pair(
+                    dia_object.get('id'), dia_source.getId())
+        self.commit()
+
+    @pipeBase.timeMethod
+    def store_updated(self, dia_collection, updated_indices):
+        """ Store new DIAObjects and sources in the sqlite database.
+
+        Parameters
+        ----------
+        dia_collection : lsst.ap.association.DIAObjectCollection
+        updated_indices : int ndarray
+            Indices within the DIAObjctCollection that contain DIAObjects
+            that were updated by AssociationTask.
+        """
```

Comment:
So there's a bit of confusion here. When I say "updated_indices" I index in the set of DIAObjects that make up the DIAObjectCollection. It has nothing to do with the indexer indices which I agree is a bit confusing. I've changed the input name to updated_dia_collection_indices for now, but one proposal is to change the input to DIAObject ids that have been updated instead to avoid confusion.

Reply:
I think that makes sense? Assuming you don't mean the Object IDs are subject to change (which will be very confusing for science users).

Reply:
The object id's are not subject to change, only the summary statics that represent that are stored in the DIAObject and hence the need to store them when the DIAObject has a source appended to it. For clarity, should I keep updated_dia_collection_indices or should I move forward with specifying the id's of the DIAObjects that need to be updated?

Reply:
IDs would definitely be clearer.